test/unit/select_dropped_indexes_test.lua

```diff
@@ -0,0 +1,120 @@
+local select_plan = require('crud.select.plan')
+
+local select_conditions = require('crud.select.conditions')
+local cond_funcs = select_conditions.funcs
+
+local t = require('luatest')
+local g = t.group('select_dropped_indexes')
+
+local helpers = require('test.helper')
+
+g.before_all = function()
+    helpers.box_cfg()
+
+    local customers = box.schema.space.create('customers', {
+        format = {
+            {'id', 'unsigned'},
+            {'bucket_id', 'unsigned'},
+            {'name', 'string'},
+            {'age', 'unsigned'},
+            {'number_of_pets', 'unsigned'},
+            {'cars', 'array'},
+        },
+        if_not_exists = true,
+    })
+
+    customers:create_index('index1', {
+        type = 'TREE',
+        parts = {'id'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('index2', {
+        parts = {'bucket_id'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers:create_index('index3', {
+        type = 'TREE',
+        parts = {'age'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers:create_index('index4_dropped', {
+        type = 'HASH',
+        parts = {'name'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('index5_dropped', {
+        type = 'HASH',
+        parts = {'age'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('index6', {
+        type = 'TREE',
+        parts = {'name'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers.index.index4_dropped:drop()
+    customers.index.index5_dropped:drop()
+end
+
+g.after_all = function()
+    box.space.customers:drop()
+end
+
+g.test_conditions = function()
+    t.assert(#box.space.customers.index ~= table.maxn(box.space.customers.index))
```

Comment:
I don't understand such test, @olegrok, was it your suggestion?

Reply:
yes, it was my suggestion

Reply:
I need a comment to understand why do we need such test.

Moreover, I sure that it shouldn't be a separated test. I think that such assertions should be performed for all test group (I mean `before_all` section), and should lead to fail of all group tests.

Reply:
My suggestion was just add a check. It's needed to check initial conditions correctness that test actually covers a problem. Yes, I agree that assert in before_all should be enough.